backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,502 @@
+	}
+
+	if pipeline.Namespace == "" {
+		if common.IsMultiUserMode() || common.GetPodNamespace() == "" {
+			return nil, util.NewBadRequestError(errors.New("A namespace is required"), "")
+		}
+
+		pipeline.Namespace = common.GetPodNamespace()
+	}
+
+	pipeline.Status = model.PipelineCreating
+
+	k8sPipeline := v2beta1.FromPipelineModel(*pipeline)
+
+	glog.Infof("Creating the pipeline %s/%s in Kubernetes", k8sPipeline.Namespace, k8sPipeline.Name)
+
+	err := k.client.Create(context.TODO(), &k8sPipeline)
+	if k8serrors.IsAlreadyExists(err) {
+		return nil, util.NewAlreadyExistError(
+			"Failed to create a new pipeline. The name %v already exists. Please specify a new name", pipeline.Name,
+		)
+	} else if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create the pipeline")
+	}
+
+	err = k.client.Status().Update(context.TODO(), &k8sPipeline)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to set the pipeline status")
+	}
+
+	return k8sPipeline.ToModel(), nil
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineStatus(pipelineId string, status model.PipelineStatus) error {
+	// Do nothing. Just return nil to avoid show an unrelated error
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineVersionStatus(pipelineVersionId string, status model.PipelineVersionStatus) error {
+	k8sPipelineVersion, err := k.getK8sPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return err
+	}
+
+	conditionSet := false
+
+	for i := range k8sPipelineVersion.Status.Conditions {
+		condition := &k8sPipelineVersion.Status.Conditions[i]
+
+		if condition.Type == "PipelineVersionStatus" {
+			if condition.Reason == string(status) && condition.Message == condition.Reason {
+				return nil
+			}
+
+			condition.Reason = string(status)
+			condition.Message = string(status)
+			condition.Status = metav1.ConditionTrue
+
+			conditionSet = true
+
+			break
+		}
+	}
+
+	if !conditionSet {
+		k8sPipelineVersion.Status.Conditions = append(k8sPipelineVersion.Status.Conditions, v2beta1.SimplifiedCondition{
+			Type:    "PipelineVersionStatus",
+			Reason:  string(status),
+			Message: string(status),
+			Status:  metav1.ConditionTrue,
+		})
+	}
+
+	// TODO: Add resilience to resource version changing
+	err = k.client.Status().Update(context.TODO(), k8sPipelineVersion)
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline version status")
+	}
+
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineVersion(pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	pipeline, err := k.GetPipeline(pipelineVersion.PipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.createPipelineVersionWithPipeline(pipeline, pipelineVersion)
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
```

Comment:
why did we opt not to support this?

Reply:
thanks @mprahl 

@rimolive can we get a inline code comment explaining the reasoning

Reply:
I added a comment as suggested